### PR TITLE
feat: not eoa

### PR DESCRIPTION
**Pull request by ashitakah**
# 🤖 Linear

Closes GRT-21

### CODE REVIEW

File: solidity/contracts/modules/finality/CallbackModule.sol [hunk 1]
```diff
@@ -35,11 +36,24 @@ contract CallbackModule is Module, ICallbackModule {
...
   function validateParameters(bytes calldata _encodedParameters)
     external
-    pure
+    view
     override(Module, IModule)
     returns (bool _valid)
   {
     RequestParameters memory _params = decodeRequestData(_encodedParameters);
-    _valid = address(_params.target) != address(0) && _params.data.length != 0;
+    _valid = address(_params.target) != address(0) && _params.data.length != 0 && _targetHasBytecode(_params.target);
```
Comment (0xJabberwock) [inline solidity/contracts/modules/finality/CallbackModule.sol:44]: Can `address(0)` ever have bytecode? If not, the first comparison could be avoided.

File: solidity/contracts/modules/finality/CallbackModule.sol [hunk 2]
```diff
@@ -27,6 +27,7 @@ contract CallbackModule is Module, ICallbackModule {
     address _finalizer
   ) external override(Module, ICallbackModule) onlyOracle {
     RequestParameters memory _params = decodeRequestData(_request.finalityModuleData);
+
     _params.target.call(_params.data);
     emit Callback(_response.requestId, _params.target, _params.data);
     emit RequestFinalized(_response.requestId, _response, _finalizer);
```
Comment (0xJabberwock) [inline solidity/contracts/modules/finality/CallbackModule.sol:33]: How come `validateParameters()` gets called during this context?

**Review by 0xJabberwock — CHANGES_REQUESTED** [review]
The audit mentions that the issue is also present at least in `CircuitResolver.sol`.

File: solidity/contracts/modules/finality/CallbackModule.sol (hunk 2 above)
Comment (0xShaito) [reply solidity/contracts/modules/finality/CallbackModule.sol:33]: It's an off-chain validation

**Review by 0xJabberwock — APPROVED** [review]

**Review by 0xShaito — APPROVED** [review]

**Pull request merged**